File: python/app/services/report_service.py
import asyncio
from typing import Optional, Dict, Any, List
from app.core.task_manager import TaskManager
from app.core.workflow_logger import AIWorkflowLogger, StepType
from app.agents.report_agent import ReportAgent


class ReportGenerationService:
    def __init__(self, task_manager: TaskManager, workflow_logger: AIWorkflowLogger, config: Any):
        self.task_manager = task_manager
        self.logger = workflow_logger
        self.config = config
        self.report_agent = ReportAgent(task_manager, workflow_logger, config)
        self._reports: Dict[str, Dict] = {}

    async def generate_react_report(self, task_id: str, process_task_id: Optional[str] = None) -> str:
        report = await self.report_agent.generate_report(task_id, process_task_id)
        self._reports[task_id] = {
            "report": report,
            "reasoning_steps": self.report_agent.get_reasoning_steps()
        }
        return report

    def get_report(self, task_id: str) -> Optional[Dict]:
        return self._reports.get(task_id)

    def get_reasoning_steps(self, task_id: str) -> Optional[List[Dict]]:
        report = self._reports.get(task_id)
        return report.get("reasoning_steps") if report else None


report_service = ReportGenerationService

File: src/services/reportService.ts
import axios from 'axios'
import { DEFAULT_URLS } from '@/constants'

export interface ReACTStep {
  step_number: number
  step_type: 'thought' | 'action' | 'observation' | 'final_answer'
  content: string
  tool_name: string | null
  tool_input: Record<string, unknown> | null
  tool_output: Record<string, unknown> | null
  duration_ms: number | null
}

export interface Report {
  report: string
  reasoning_steps: ReACTStep[]
  total_steps: number
}

export interface ReportMetadata {
  task_id: string
  generated_at: string
  model: string
  total_duration_ms: number
  total_tokens: number
}

class ReportService {
  private baseUrl: string

  constructor(baseUrl: string = DEFAULT_URLS.PYTHON_BACKEND) {
    this.baseUrl = baseUrl.replace(/\/$/, '')
  }

  async generateReport(processTaskId?: string): Promise<{ task_id: string; status: string; message: string }> {
    const response = await axios.post(`${this.baseUrl}/api/v1/reports/generate`, {
      process_task_id: processTaskId || null
    })
    return response.data
  }

  async getReport(taskId: string): Promise<Report> {
    const response = await axios.get(`${this.baseUrl}/api/v1/reports/${taskId}`)
    return response.data.data
  }

  async getReasoningSteps(taskId: string): Promise<ReACTStep[]> {
    const response = await axios.get(`${this.baseUrl}/api/v1/reports/${taskId}/reasoning`)
    return response.data.data.reasoning_steps
  }
}

export const reportService = new ReportService()
